### PR TITLE
Replace `thrust::complex` with `cuda::std::complex`

### PR DESCRIPTION
**Pull request by miscco**
There are some notable differences though. thrust::complex has been a
bit more lenient when determining the type of arithmetic operations.

That said, I believe being more strict is actually a feature not a bug

### CODE REVIEW

File: thrust/thrust/detail/complex/arithmetic.h [hunk 1]
```diff
@@ -1,303 +0,0 @@
...
-
-
-template <typename T0, typename T1>
-__host__ __device__
-complex<typename detail::promoted_numerical_type<T0, T1>::type>
-operator-(const complex<T0>& x, const complex<T1>& y)
-{
-  typedef typename detail::promoted_numerical_type<T0, T1>::type T;
-  return complex<T>(x.real() - y.real(), x.imag() - y.imag());
-}
```
Comment (senior-zero) [inline thrust/thrust/detail/complex/arithmetic.h:66]: The following code used to work, but now it doesn't:
```cpp
  thrust::complex<float> a;
  thrust::complex<double> b;
  auto c = a + b;
```
I think we need to keep this overload and maybe mark it as deprecated. Same with other operators.

File: thrust/thrust/detail/complex/arithmetic.h [hunk 2]
```diff
@@ -1,303 +0,0 @@
...
-
-template <typename T0, typename T1>
-__host__ __device__
-complex<typename detail::promoted_numerical_type<T0, T1>::type>
-operator+(const T0& x, const complex<T1>& y)
-{
-  typedef typename detail::promoted_numerical_type<T0, T1>::type T;
-  return complex<T>(x + y.real(), y.imag());
-}
```
Comment (senior-zero) [inline thrust/thrust/detail/complex/arithmetic.h:56]: The following code used to work, but now it doesn't:

```cpp
  float a = 84;
  thrust::complex<double> b(42, 42);
  auto c = a + b;
```
I think we need to keep this overload and maybe mark it as deprecated. Same with other operators.

File: thrust/thrust/detail/complex/arithmetic.h [hunk 3]
```diff
@@ -1,303 +0,0 @@
...
-  return y * -T(1);
-}
-
-
-/* --- Other Basic Arithmetic Functions --- */
-
-// As std::hypot is only C++11 we have to use the C interface
-template <typename T>
-__host__ __device__
-T abs(const complex<T>& z)
```
Comment (senior-zero) [inline thrust/thrust/detail/complex/arithmetic.h:186]: The following code used to work, now it doesn't:
```cpp
  thrust::complex<double> a;
  auto b = thrust::abs(a);
```

File: thrust/thrust/detail/complex/arithmetic.h [hunk 4]
```diff
@@ -1,303 +0,0 @@
...
-__host__ __device__
-inline double abs(const complex<double>& z)
-{
-  return detail::complex::abs(z);
-}
-
-
-template <typename T>
-__host__ __device__
-T arg(const complex<T>& z)
```
Comment (senior-zero) [inline thrust/thrust/detail/complex/arithmetic.h:227]: The following code used to work, now it doesn't:
```cpp
  thrust::complex<double> a;
  auto b = thrust::arg(a);
```

**Review by senior-zero — CHANGES_REQUESTED** [review]
I've left a few comments on places that break API. I haven't covered all the changes, but I think that the comments should be applicable to the rest of them.

**Comment by miscco** [comment]
Dropping in favor of #454

**Pull request closed without merging**